### DISCUSSION TITLE
Fabric.js : Text not render correct with charSpacing  in safari browser.

### DISCUSSION THREAD

**Discussion by niketaOptimumnbrew** [fabricjs/fabric.js#9908] [General]
Hey,

Fabric.js version 5.3.0

I have encountered one issue with text in safari browser , that text's gets cut off when charSpacing is 0 and  font Weight is 'bold'.

The following is my code:
////////////////////////////////////////////////////////////////////////////
var canvas = new fabric.Canvas("canvas");
fabric.Object.prototype.objectCaching = false;

var font_family = "Roboto-Regular";
var font = new FontFaceObserver(font_family);

var Text = new fabric.Textbox("ADD HEADING TEXT", {
left: 210,
top: 261,
fontSize: 69,
width: 630,
textAlign: 'left',
fontWeight:'bold',
fill: 'black',
charSpacing:0,
strokeWidth: 0,
      stroke: '#ffffff',
      shadow: {
        "color": "#000000",
        "blur": 0,
        "offsetX": 0,
        "offsetY": 0,
        "affectStroke": false
      },
hasRotatingPoint: true,
      isLocked: false,
      shadow_theme_index: undefined,
});

font.load().then(function() {
fabric.util.clearFabricFontCache(font_family);
Text.set("fontFamily", font_family);
canvas.requestRenderAll();
});

canvas.add(Text);
////////////////////////////////////////////////////////////////////////////

Please check below image (safari browser):
[
![image](https://github.com/fabricjs/fabric.js/assets/126561906/a3d2e121-b5ce-4b44-9492-6c647e4de01a)
](url)

Text should show within the bounding box, Like below image (chrome browser):
![image](https://github.com/fabricjs/fabric.js/assets/126561906/b7c17f6e-51f2-452c-a41a-e03da7c30975)

**Comment by mhieupham1** [comment]
any jsfiddle for example code ? i think if you use jsfiddle to show everyone, it easier to hotfix @niketaOptimumnbrew

**Reply by niketaOptimumnbrew** [reply]
@mhieupham1 
Please check below fiddle :
https://jsfiddle.net/Niketa_patel/6rbq4Lz8/25/

**Reply by mhieupham1** [reply]
@niketaOptimumnbrew i can't explain why, but when i calc the width and height of the text without using cache, it gives correct result 

try https://jsfiddle.net/50e6t879/2/
